cfg/common.go

```diff
@@ -167,6 +167,28 @@ type Common struct {
 
 	Traces               Traces `yaml:"traces"`
 	PrintErrorStackTrace bool   `yaml:"printErrorStackTrace"`
+
+	// DeterministicBackend sets a backend weight for each metrics in render
+	// request and carbonzipper would favor metrics from a specific node.
+	// As storage backends like go-carbon doesn't have strong consistency
+	// guarantee, this means that the same metric persisted in different
+	// replicas might have different values. Without DeterministicBackend
+	// option, carbonzipper would just favor metrics returned first from
+	// any backend. This might result in confusing query result in
+	// carbonapi that are reading multiple metrics in one query.
+	//
+	// An example like bellow:
+	//
+	//     asPercent(
+	//         integral(sumSeries(http.app.latencies.{0_5_1,1_5,5_10,10_25,25_50,50_100,100_250}.count)),
+	//         integral(sumSeries(http.app.latencies.*.count))
+	//     )
+	//
+	// Without DeterministicBackend option, the example query above might
+	// return result 100.1%. With DeterministicBackend option, it's
```

Comment:
Not for this example. Reword the comments to make it clear:

> Without DeterministicBackend option, if there are inconsistent values
> between replicas in the returning metrics (for example, in one
> replica 0_5_1.count has value 10, and in another 0_5_1.count has 5),
> the example query above might return percentage values larger than
> 100%. With DeterministicBackend option, it's guaranteed that the
> query result would always "seem" meaningful (in the example, it
> always return value <= 100%), but not necessarily correct.

Reply:
Hmm, sorry, after rephrase it didn't get any clearer for me 😬

Reply:
```
asPercent(
    integral(sumSeries(http.app.latencies.{0_5_1,1_5,5_10,10_25,25_50,50_100,100_250}.count)),
    integral(sumSeries(http.app.latencies.*.count))
)
```

Let's suppose there are two replicas and only metric `0_5_1.count` has inconsistent values at `timestamp T` (`10` in replica a, `5` in replica b).

Without `DeterministicBackend`, carbonzipper might fetches `0_5_1.count` for `integral(sumSeries(http.app.latencies.{0_5_1,1_5,5_10,10_25,25_50,50_100,100_250}.count)),` from replica a, and fetches `0_5_1.count` for `integral(sumSeries(http.app.latencies.*.count))` from replica b.

These means that at `timestamp T`, `integral(sumSeries(http.app.latencies.{0_5_1,1_5,5_10,10_25,25_50,50_100,100_250}.count))` is larger than `integral(sumSeries(http.app.latencies.*.count))`, and it would produces a percentage value >= 100%.

With `DeterministicBackend`, carbonzipper would either use replica a or replica b for both subqueries thus its guarantee to return a percentage value <= 100%, which is arguably more meaningful.

Reply:
ohh, yep got it now, thx